src/test/java/com/scottlogic/filters/AuthorPostFilterTest.java

```diff
@@ -0,0 +1,86 @@
+package com.scottlogic.filters;
+
+import com.scottlogic.UserPost;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class AuthorPostFilterTest {
+    UserPost userPost1 = new UserPost("Jane Smith",
+            OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Hello World!", 2);
+
+    UserPost userPost2 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    UserPost userPost3 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks.", 3);
+
+    @Test
+    void filter_null_null() {
+        List<UserPost> actual = new AuthorPostFilter("").filter(null);
+
+        assertNull(actual);
+    }
+
+    @Test
+    void filter_EmptyList_EmptyList() {
+        List<UserPost> inputList = Collections.emptyList();
+        List<UserPost> expected = Collections.emptyList();
+
+        List<UserPost> actual = new AuthorPostFilter("").filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void filter_1ItemList_1ItemList() {
```

Comment:
Could be good to do a singular item that gets filtered out completely? Although it is covered by the 3 items to none it is at a different case level, this being the base case, so it could be worth adding it in despite this.